app/uk/co/panaxiom/playjongo/PlayJongo.java

```diff
@@ -41,20 +42,28 @@ public PlayJongo(ApplicationLifecycle lifecycle, Environment env, Configuration
     }
 
     PlayJongo(Configuration config, ClassLoader classLoader, boolean isTestMode) throws Exception {
-        configure(config,classLoader,isTestMode);
+        configure(config, classLoader, isTestMode);
     }
 
     private void configure(Configuration config, ClassLoader classLoader, boolean isTestMode) throws Exception {
-        
-        String clientFactoryName = config.getString("playjongo.mongoClientFactory");
-        MongoClientFactory factory = getMongoClientFactory(clientFactoryName, config, isTestMode);
-        mongo = factory.createClient();
 
-        if (mongo == null) {
-            throw new IllegalStateException("No MongoClient was created by instance of "+ factory.getClass().getName());
-        }
+        DB db;
+        if (config.getBoolean("playjongo.useFongo", false)) {
+
+            db = new Fongo("in-memory").getDB(config.getString("playjongo.fongo.db", "fongodb"));
+
+        } else {
 
-        DB db = mongo.getDB(factory.getDBName());
+            String clientFactoryName = config.getString("playjongo.mongoClientFactory");
+            MongoClientFactory factory = getMongoClientFactory(clientFactoryName, config, isTestMode);
+            mongo = factory.createClient();
+
+            if (mongo == null) {
+                throw new IllegalStateException("No MongoClient was created by instance of " + factory.getClass().getName());
+            }
+
+            db = mongo.getDB(factory.getDBName());
+        }

```

Comment:
It would probably be better to extract this block to a protected method so that it can be overridden by a class to be used by tests.